model/user_autocomplete.go

```diff
@@ -17,6 +17,22 @@ type UserAutocompleteInTeam struct {
 	InTeam []*User `json:"in_team"`
 }
 
+type UserAutocomplete struct {
+	InSystem     []*User `json:"in_system"`
```

Comment:
Let's make the first one just `Users` and that will be the main field that always gets populated. For example if we're searching the whole system `Users` gets populated with those users, if we're searching just in a channel then `Users` gets populated in those cases.

The only other fields will be for if we need to return extra autocomplete information. Right now I think that's only `OutOfChannel` so the two fields we should have would be `Users` and `OutOfChannel`.

Could you also omit `OutOfChannel` if it's empty by changing `json:"out_of_channel"` to `json:"out_of_channel, omitempty"`?